src/main/scala/nl.knaw.dans.easy.dd2d/DepositIngestTask.scala

```diff
@@ -82,29 +80,26 @@ case class DepositIngestTask(deposit: Deposit, dansBagValidator: DansBagValidato
     // TODO: delete draft if something went wrong
   }
 
-  private def formatViolation(v: (String, String)): String = v match {
-    case (nr, msg) => s" - [$nr] $msg"
+  private def getPersistentId(response: DataverseResponse[DatasetCreationResult]): Try[String] = Try {
+    response.data.get.persistentId.toString
```

Comment:
response.data gives you a Try and you need a Try in the return value of getPersistentId; then why call `get`? You can just do

```
    response.data.map(_.persistentId)
```